src/richie/apps/courses/api.py

```diff
@@ -159,18 +159,19 @@ def sync_course_run(data):
 
     # Look for the course targeted by the resource link
     course_code = normalize_code(lms.extract_course_code(data))
-    try:
-        course = Course.objects.get(
+    course = (
+        Course.objects.filter(
             code=course_code,
             extended_object__publisher_is_draft=True,
             # Exclude snapshots
             extended_object__node__parent__cms_pages__course__isnull=True,
         )
-    except Course.DoesNotExist as exc:
+        # distinct fixes MySQL returning same course twice
```

Comment:
I don't think it is related to mysql, but to the join in the clause:
`extended_object__node__parent__cms_pages`
```suggestion
```

Reply:
Yes, it also works! passed the tests 😀. So I'm changing to your solution.